### DISCUSSION TITLE
go:generate compile errors on core Go modules

### DISCUSSION THREAD

**Discussion by mpetronic** [golang/vscode-go#2056] [Discussions]
Can anyone explain why I am getting errors like this trying to generate a mock for an interface? If I include the "io" package, it fails. If I do not include the "io" package, it works. Here is a simple example Go module that I am testing with that reproduces the errors.
```
// THIS FAILS WITH COMPILE ERRORS

package simple

import (
	"bytes"
	"io"
)

//go:generate mockgen -source=simple.go -destination=mock_simple.go -package=simple

type ISimple interface {
	Get() io.Reader
}

var _ ISimple = (*Simple)(nil)

type Simple struct{}

func (s *Simple) Get() io.Reader {
	return bytes.NewReader([]byte{123})
}
```
The errors I get are like this and there are very many - this is just the head of them...
```
go generate -v -x 2>&1 | head
simple.go
mockgen -source=simple.go -destination=mock_simple.go -package=simple
/usr/local/go/src/runtime/proc.go:5664:23: invalid operation: shift count (id % 32) (value of type int32) must be unsigned integer
/usr/local/go/src/runtime/proc.go:5657:23: invalid operation: shift count (id % 32) (value of type int32) must be unsigned integer
/usr/local/go/src/runtime/panic.go:826:20: invalid operation: shift count i (variable of type int) must be unsigned integer
/usr/local/go/src/runtime/panic.go:848:34: invalid operation: shift count i (variable of type int) must be unsigned integer
/usr/local/go/src/runtime/panic.go:796:26: invalid operation: shift count shift (variable of type int) must be unsigned integer
/usr/local/go/src/runtime/panic.go:798:31: invalid operation: shift count shift (variable of type int) must be unsigned integer
/usr/local/go/src/runtime/mpallocbits.go:172:9: invalid operation: shift count sys.TrailingZeros64(x) & 63 (value of type int) must be unsigned integer
/usr/local/go/src/unicode/utf8/utf8.go:29:8: expected ';', found b00000000
```

If I rework the code to NOT include "io", it works. I have tried Go 16 and 17 latest - same results. I am running latest gopls server in vscode on Windows 10 WSL. It seems like the "io" package is pulling in some dependencies that, for some bizarre reason, are resulting compile errors on core Go language modules -very strange and puzzling. I am totally stumped on this one. 

```
// THIS WORKS FINE

package simple

//go:generate mockgen -source=simple.go -destination=mock_simple.go -package=simple

type ISimple interface {
	Get() []byte
}

var _ ISimple = (*Simple)(nil)

type Simple struct{}

func (s *Simple) Get() []byte {
	return []byte{123}
}
```

**Comment by hyangah** [comment]
It looks like a problem of `mockgen`.  (gopls or even go generate has nothing to do with the error).
I'd first try to recompile/update `mockgen` with the go version you are using to make sure this is not a go version mismatch issue or an already fixed issue.  If that doesn't fix, then ask for help in the `mockgen` repo.

**Comment by mpetronic** [comment]
@hyangah - Thanks for chiming in. I was able to solve the issue. See: https://github.com/golang/mock/issues/619
